docs/api-conventions.asciidoc

```diff
@@ -32,106 +35,98 @@ Namespace clients are very lightweight objects that can be created on the fly.
-        .index("my-index")
-        .putAliases("foo",
-            new Alias.Builder().isWriteIndex(true).build()
-        )
-        .build()
-);
+ElasticsearchClient client = ...
+include-tagged::{doc-tests}/ApiConventionsTest.java[builders]
 --------------------------------------------------
 
 Note that a builder should not be reused after its `build()` method has been 
 called.
 
 Although this works nicely, having to instantiate builder classes and call the 
-build() method is a bit verbose. So every builder setter in the Java client also 
+`build()` method is a bit verbose. So every property setter in the {java-client} also
 accepts a lambda expression that takes a newly created builder as a parameter 
-and returns a populated builder. The snippet above can be written also as:
+and returns a populated builder. The snippet above can also be written as:
 
 ["source","java"]
 --------------------------------------------------
-CreateResponse createResponse = client.indices()
-    .create(createBuilder -> createBuilder
-        .index("my-index")
-        .putAliases("foo", aliasBuilder -> aliasBuilder
-    		.isWriteIndex(true)
-        )
-    );
+ElasticsearchClient client = ...
+include-tagged::{doc-tests}/ApiConventionsTest.java[builder-lambdas]
 --------------------------------------------------
 
 This approach allows for much more concise code, and also avoids importing 
 classes (and even remembering their names) since types are inferred from the 
 method parameter signature.
 
-It becomes particularly useful with complex nested queries like the one below, 
-taken from the 
+Note in the above example that builder variables are only used to start a chain
+of property setters. Their name therefore doesn't matter much and can be shortened
+to improve readability:
+
+["source","java"]
+--------------------------------------------------
+ElasticsearchClient client = ...
+include-tagged::{doc-tests}/ApiConventionsTest.java[builder-lambdas-short]
+--------------------------------------------------
+
+
+Builder lambdas become particularly useful with complex nested queries like the
+one below, taken from the
 {ref}/query-dsl-intervals-query.html[intervals query API documentation].
 
 This example also shows a useful naming convention for builder parameters in 
 deeply nested structures: since we have to give them a name to comply with the 
-Java syntax (Kotlin would accept `it` and Scala a simple `_`), we name them with 
-an underscore followed by the depth of the item, i.e. `_0`, `_1`, and so on. 
-This removes the need for finding names and makes reading the code easier to 
-read by reducing the number of identifiers.
+Java syntax, we name them with an underscore followed by the depth of the item,
+i.e. `_0`, `_1`, and so on. This removes the need for finding names and makes
+the code easier to read by reducing the number of identifiers. Correct indentation
+also allows the structure of the query to stand out.
```

Comment:
"This example also highlights a useful and recommended naming convention for builder parameters in deeply nested structures. For lambda arguments, Kotlin provides `it` and Scala allows use of `_`; this can be approximated in Java by using an underscore prefix followed by a number representing the depth level (i.e. `_0`, `_1`, and so on). Not only does this remove the need to generate throw-away variable names, but it also improves code readability."